packages/did/src/Did.signature.ts

```diff
@@ -75,13 +75,16 @@ export async function verifyDidSignature({
   const { fragment } = parse(inputUri)
   if (!fragment)
     throw new SDKErrors.SignatureMalformedError(
-      `Signature key URI "${signature.keyUri}" invalid`
+      `Signature key URI "${inputUri}" invalid`
     )
 
   const keyId: UriFragment = fragment
   const keyUri: DidResourceUri = inputUri
 
+  // Cannot simply use resolveKey here as we need to check the key relationship.
+  // Shall we add this feature in the resolveKey and remove the duplicated code below?
```

Comment:
An example of this would be in 598c249d. Simplifies quite a few other things, I’d say. Also makes the `resolve()` effectively unused.

Reply:
I agree this would be a common use case. We should either cater for it in resolveKey, or have this feature in `Did.getKey()`